Slider stepValue < 1 gives rounding errors

<!-- Checked checkbox should look like this: [x] -->
- [x ] This is not a v0.x issue. <!-- (v0.x is no longer maintained) -->
- [x ] I have searched the [issues](https://github.com/mui-org/material-ui/issues) of this repository and believe that this is not a duplicate.

## Expected Behavior 🤔
Setting my slider with a stepValue < 1 (like 0.1) should only give values that are multiplication of the stepValue

## Current Behavior 😯
Some stepValues have rounding errors like 0.7000000000000001

## Steps to Reproduce 🕹

See https://codesandbox.io/s/create-react-app-ehgqw


## Your Environment 🌎
Windows 10 18.03
Node 8.15.1

| Tech         | Version |
|--------------|---------|
| Material-UI  | v4.1.0  |
| React        |  16.8.6       |
| Browser      | latest Chrome and Firefox      |
| TypeScript   |   -      |
| etc.         |         |

I wrote a simple one

```diff --git a/packages/material-ui-lab/src/Slider/Slider.test.js b/packages/material-ui-lab/src/Slider/Slider.test.js
index 75b9ae43d..3c780bb8d 100644
--- a/packages/material-ui-lab/src/Slider/Slider.test.js
+++ b/packages/material-ui-lab/src/Slider/Slider.test.js
@@ -265,6 +265,13 @@ describe('<Slider />', () => {
       thumb.simulate('keydown', moveRightEvent);
       assert.strictEqual(wrapper.props().value, 30);
     });
+
+    it('should round value to step precision', () => {
+      wrapper.setProps({ value: 0.2, step: 0.1, min: 0 });
+      const thumb = findThumb(wrapper);
+      thumb.simulate('keydown', moveRightEvent);
+      assert.strictEqual(wrapper.props().value, 0.3);
+    });
   });
 
   describe('markActive state', () => {
```

Which throws `AssertionError: expected 0.30000000000000004 to equal 0.3` before the fix.

Do you feel this test adds any value?

--- a/packages/material-ui-lab/src/Slider/Slider.js
+++ b/packages/material-ui-lab/src/Slider/Slider.js
@@ -67,8 +67,15 @@ function percentToValue(percent, min, max) {
   return (max - min) * percent + min;
 }
 
+function ensurePrecision(value, step) {
+  const stepDecimalPart = step.toString().split('.')[1];
+  const stepPrecision = stepDecimalPart ? stepDecimalPart.length : 0;
+
+  return Number(value.toFixed(stepPrecision));
+}
+
 function roundValueToStep(value, step) {
-  return Math.round(value / step) * step;
+  return ensurePrecision(Math.round(value / step) * step, step);
 }
 
 function setValueIndex({ values, source, newValue, index }) {

--- a/packages/material-ui-lab/src/Slider/Slider.test.js
+++ b/packages/material-ui-lab/src/Slider/Slider.test.js
@@ -265,6 +265,13 @@ describe('<Slider />', () => {
       thumb.simulate('keydown', moveRightEvent);
       assert.strictEqual(wrapper.props().value, 30);
     });
+
+    it('should round value to step precision', () => {
+      wrapper.setProps({ value: 0.2, step: 0.1, min: 0 });
+      const thumb = findThumb(wrapper);
+      thumb.simulate('keydown', moveRightEvent);
+      assert.strictEqual(wrapper.props().value, 0.3);
+    });
   });
 
   describe('markActive state', () => {

